core/src/test/java/io/ably/lib/test/realtime/RealtimeAuthTest.java

```diff
@@ -956,4 +963,92 @@ public Object getTokenRequest(Auth.TokenParams params) throws AblyException {
         }
     }
 
+    /**
+     * Verify that instance of ClientOptions cannot change AblyRealtime options once it is provided to constructor
+     */
+    @Test
+    public void auth_client_options_immutable() {
+        try {
+            /* create token with clientId */
+            ClientOptions optsForToken = createOptions(testVars.keys[0].keyStr);
+            optsForToken.clientId = "token_clientId";
+            AblyBase<PushBase, Platform, RestChannelBase> ablyForToken = createAblyRest(optsForToken);
+            TokenDetails tokenDetails = ablyForToken.auth.requestToken(null, null);
+
+            /* create ably realtime */
+            ClientOptions opts = new ClientOptions();
+            opts.clientId = null;
+            opts.token = tokenDetails.token;
+            opts.autoConnect = false;
+            opts.headers = new HashMap<>();
+            opts.headers.put("old_key", "old_value");
+            opts.tokenDetails = new TokenDetails("my_old_details_token");
+            opts.logHandler = new Log.DefaultHandler();
+            opts.authCallback = new Auth.TokenCallback() {
+                @Override
+                public Object getTokenRequest(Auth.TokenParams params) {
+                    return null;
+                }
+            };
+            opts.authHeaders = new Param[1];
+            opts.authHeaders[0] = new Param("old_key", "old_key");
+            opts.proxy = new ProxyOptions();
+            opts.proxy.host = "https://ably.com";
+            opts.proxy.port = 8080;
+            AblyRealtimeBase<PushBase, Platform, RealtimeChannelBase> ablyRealtime = createAblyRealtime(opts);
+
+            assertEquals("clientId should be equal before change", opts.clientId, ablyRealtime.options.clientId);
+            opts.clientId = "my_new_clientId";
+            assertNotEquals("clientId should not be equal after change", opts.clientId, ablyRealtime.options.clientId);
```

Comment:
```suggestion
            assertNotEquals("'clientId's should not be equal after change", opts.clientId, ablyRealtime.options.clientId);
```

Reply:
As clientId in question is a variable I do not think it is necessary to add an apostrophe as this is a singular object.

Reply:
I have changed text of the test to be more descriptive, see if it fits here https://github.com/ably/ably-java/pull/827/commits/e9fa9409729ad4ea6b925f6fea118e9afbbfca24